src/server/api/routers/user.ts

```diff
@@ -26,4 +26,15 @@ export const userRouter = createTRPCRouter({
   getAll: protectedProcedure.query(async () => {
     return await db.user.findMany();
   }),
+
+  getByRole: protectedProcedure.input(z.string()).query(async ({ input }) => {
+    return await db.profile.findMany({
```

Comment:
Async functions return promises and tRPC will await any promise we return, so we don't need to await the promise returned by Prisma here before returning in this function. The behavior will be identical for both ways, with or without the await here, but we don't need the await so better to leave it out.

Thinking about it, depending on the JS runtime implementation, awaiting here might cost more memory as an intermediate promise object might be made here and resolved before returning. Either way you can drop the await and it still works. :D